src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -478,4 +485,41 @@ public void updateFlowFrameworkSystemIndexDocWithScript(
             }
         }
     }
+
+    /**
+     * Creates a new ResourceCreated object and a script to update the state index
+     * @param workflowId workflowId for the relevant step
+     * @param nodeId WorkflowData object with relevent step information
+     * @param workflowStepName the workflowstep name that created the resource
+     * @param resourceId the id of the newly created resource
+     * @param completableFuture the CompletableFuture used for this step
+     * @throws IOException if parsing fails on new resource
+     */
+    public void updateResourceInStateIndex(
+        String workflowId,
+        String nodeId,
+        String workflowStepName,
+        String resourceId,
+        CompletableFuture<WorkflowData> completableFuture
```

Comment:
Instead of a `CompletableFuture` here, maybe we can use an action listener instead. Then within each `WorkflowStep` we can complete `onResponse` or `CompleteExceptionally` on failure.

Reply:
So in each workflow step, the updates would look something like : 
```java
flowFrameworkIndicesHandler.updateResourceInStateIndex(
                        currentNodeInputs.getWorkflowId(),
                        currentNodeId,
                        getName(),
                        mlCreateConnectorResponse.getConnectorId(),
                        ActionListener.wrap(
                            onResponse -> {
                                future.complete(WorkflowData);
                            },
                            onFailure -> {
                                future.completeExceptionally(onFailure.getMessage());
                            }
                        )
                    );
```